lib/Storage/MySqlTimeSeriesStore.php

```diff
@@ -73,27 +77,65 @@ public function setDependencies($log, $date, $layoutFactory = null, $campaignFac
     /** @inheritdoc */
     public function addStat($statData)
     {
-        // Set to Unix Timestamp
-        foreach ($statData as $k => $stat) {
-            $statData[$k]['statDate'] = $statData[$k]['statDate']->format('U');
-            $statData[$k]['fromDt'] = $statData[$k]['fromDt']->format('U');
-            $statData[$k]['toDt'] = $statData[$k]['toDt']->format('U');
-        }
 
-        $sql = 'INSERT INTO `stat` (`type`, statDate, start, `end`, scheduleID, displayID, campaignID, layoutID, mediaID, Tag, `widgetId`, duration, `count`) VALUES ';
-        $placeHolders = '(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)';
+        // For a type "event" we have layoutid 0 so is campaignId
+        // otherwise we should try and resolve the campaignId
+        $campaignId = 0;
+        if ($statData['type'] != 'event') {
+
+            if (array_key_exists($statData['layoutId'], $this->layoutCampaignIds)) {
+                $campaignId = $this->layoutCampaignIds[$statData['layoutId']];
+            } else {
+
+                try {
+
+                    // Get the layout campaignId
+                    $campaignId = $this->layoutFactory->getCampaignIdFromLayoutHistory($statData['layoutId']);
 
-        $sql = $sql . implode(', ', array_fill(1, count($statData), $placeHolders));
+                    // Put layout campaignId to memory
+                    $this->layoutCampaignIds[$statData['layoutId']] = $campaignId;
 
-        // Flatten the array
-        $data = [];
-        foreach ($statData as $stat) {
-            foreach ($stat as $field) {
-                $data[] = $field;
+                } catch (XiboException $error) {
+
+                    $this->log->error('Layout not found. Layout Id: '. $statData['layoutId']);
```

Comment:
We used to only log this one time per layoutId - does this new code log it each time?